### PR TITLE
feat: support for DID migrations

### PR DESCRIPTION
**Pull request by Diiaablo95**
## fixes KILTProtocol/ticket#1554

The PR allows for a light DID to be upgraded to a full DID by writing on the KILT blockchain a DID with the same identifier as the light one. Furthermore, attested claim presentation verification has been modified to only accept signatures from a valid DID if a DID has been migrated since the claim was initially created and included in an attestation.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/did/src/DidDetails/LightDidDetails.ts [hunk 1]
```diff
@@ -94,43 +93,4 @@ export class LightDidDetails extends DidDetails {
       this.keyRelationships.keyAgreement = [`${this.didUri}#encryption`]
     }
   }
-
```
Comment (Diiaablo95) [inline packages/did/src/DidDetails/LightDidDetails.ts:97]: Removed as I realized they are already exposed by the `DidDetails` abstract superclass.

File: packages/did/src/DidResolver/DefaultResolver.ts [hunk 2]
```diff
@@ -215,7 +231,7 @@ export async function resolveKey(
         throw SDKErrors.ERROR_INVALID_DID_FORMAT(didUri)
       }
       // The fragment includes the '#' symbol which we do not need
-      return resolvedDetails.getKey(fragment.substring(1)) || null
+      return resolvedDetails.details.getKey(didUri) || null
```
Comment (arty-name) [inline packages/did/src/DidResolver/DefaultResolver.ts:234]: The comment above is not needed anymore

**Review by tjwelde — COMMENTED** [review]
Some questions below

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**